rfcs/text/0008-email.md

```diff
@@ -0,0 +1,144 @@
+# 0008: Email
+<!-- Leave this ID at 0000. The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC. -->
+
+- Stage: **1** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
+- Date: **Oct 5th 2020** <!-- The ECS team sets this date at merge time. This is the date of the latest stage advancement. -->
+
+This RFC proposes a new top-level field to facilitate email use cases. 
+
+<!--
+As you work on your RFC, use the "Stage N" comments to guide you in what you should focus on, for the stage you're targeting.
+Feel free to remove these comments as you go along.
+-->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+
+## Fields
+
+<!--
+Stage 1: Describe at a high level how this change affects fields. Which fieldsets will be impacted? How many fields overall? Are we primarily adding fields, removing fields, or changing existing fields? The goal here is to understand the fundamental technical implications and likely extent of these changes. ~2-5 sentences.
+-->
+
+| field | type | description |
+| --- | --- | --- |
+| `email.action` | keyword | Action take by the source device, e.g. delivered, blocked, quarantined, deleted |
+| `email.bcc.address` | keyword | Addresses of Bcc's |
+| `email.bcc.domain` | keyword | Domains of the Bcc's |
+| `email.cc.address` | keyword | Addresses of Cc's |
+| `email.cc.domain` | keyword | Domains of Cc addresses |
+| `email.cipher` | keyword | Cipher used e.g. TLS |
+| `email.file.count` | value | Number of attachments included in the message |
+| `email.file.extension` | keyword | Extensions of attachment, e.g. .zip, .docx |
+| `email.file.hash` | keyword | Hash of attachments |
+| `email.file.name` | keyword | File name of attachements |
+| `email.file.size` | keyword | Total size of all attachements in bytes |
+| `email.direction` | keyword | Direction of the message based on the sending and receving domains |
+| `email.from.address` | keyword | Senders email address |
+| `email.from.domain` | keyword | Senders domain |
+| `email.latency` | keyword | The time, in milliseconds, the delivery attempt took |
+| `email.message_id` | keyword | Internet message ID of the message |
+| `email.process` | keyword | Name of the executable that carried out the transaction, e.g. outlook, sendmail |
```

Comment:
MTA?

Reply:
Why add a field for this, i.m.o this can be captured in `process.*` fields. Specifically the `process.name` field.

This would be in line with [the statement](https://github.com/elastic/ecs/pull/999/files#r512252692) on `email.action`

Reply:
I agree with @SHolzhauer on this, I think the name of the process should follow the current ECS format.

When it comes to #2 from @webmat it is something that might be useful in a separate field.